File: src/Publisher.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include "performance_tests/SuperAwesome.h"
#include "performance_tests/Rate.h"
#include <sstream>

int Rate = 1;
int Counter  = 0;

bool Rate_Callback(performance_tests::Rate::Request  &req, performance_tests::Rate::Response &res)
{
    Rate = req.data;
    Counter  = 0;
    
	ROS_INFO("Loop rate: %d", (int) req.data);

  	return true;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "Publisher_Cpp");
    ros::NodeHandle n;

    ros::Publisher      SuperAwesome_Publisher  = n.advertise<performance_tests::SuperAwesome>("SuperAwesome_Topic", 1000);
    ros::ServiceServer  Rate_Server             = n.advertiseService("Change_Rate_Service", Rate_Callback);

    performance_tests::SuperAwesome msg;

    ros::Rate loop_rate(1);

    while (ros::ok())
    {
        std::stringstream ss;
        ss << ++Counter;
        msg.data = ss.str();
        ROS_INFO("%s", msg.data.c_str());
        SuperAwesome_Publisher.publish(msg);

        ros::spinOnce();
        loop_rate = ros::Rate(Rate);
        loop_rate.sleep();
    }

    return 0;
}

File: scripts/Publisher.py
import rospy
from performance_tests.msg import SuperAwesome
from performance_tests.srv import Rate

Counter     = 0
LoopRate    = 1

def Rate_Callback(req):
    global LoopRate
    global Counter 
    LoopRate = req.data
    Counter = 0
    print "Loop rate: %d" %(LoopRate)    
    return True

def talker():
    rospy.init_node('Publisher_Python', anonymous=True)
    SuperAwesome_Publisher  = rospy.Publisher('SuperAwesome_Topic', SuperAwesome, queue_size=10)
    Rate_Server             = rospy.Service('Change_Rate_Service', Rate, Rate_Callback)
    rate = rospy.Rate(1) 

    while not rospy.is_shutdown():
        global LoopRate
        global Counter 
        Counter += 1
        hello_str = str(Counter)
        rospy.loginfo(hello_str)
        SuperAwesome_Publisher.publish(hello_str)
        rate = rospy.Rate(LoopRate) 
        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass